patches/minecraft/net/minecraft/block/BlockRailBase.java.patch

```diff
@@ -48,8 +61,20 @@
              {
                  flag = true;
              }
-@@ -162,6 +162,98 @@
+@@ -148,7 +148,7 @@
+     {
+         super.func_180663_b(p_180663_1_, p_180663_2_, p_180663_3_);
+ 
+-        if (((BlockRailBase.EnumRailDirection)p_180663_3_.func_177229_b(this.func_176560_l())).func_177018_c())
++        if (getRailDirection(p_180663_1_, p_180663_2_, p_180663_3_, null).func_177018_c())
+         {
+             p_180663_1_.func_175685_c(p_180663_2_.func_177984_a(), this, false);
+         }
+@@ -160,8 +160,101 @@
+         }
+     }
  
++    //Forge: Use getRailDirection(IBlockAccess, BlockPos, IBlockState, EntityMinecart) for enhanced ability
```

Comment:
No, there are legitimate uses for it like you said. I this this comment is good enough.